### DISCUSSION TITLE
🔗 🚀 Feature Request: New Widget Components & Instancing System (Duplicate of #965)

### DISCUSSION THREAD

**Discussion by Henzxon** [elementor/elementor#22850] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

**Scenario:** 

> I have 4 global button variations but they all have the same links

Button 1 : Solid border; transparent background; no icon
Button 2 : No border, underlined, with icon
Button 3 : Just text, underline on hover, with icon
Button 4 : Solid border, with background, with icon

They are placed across the site varying depending on the container/section background. So each global widget might have 2 version each, one for dark background and one for light background.

### Describe the solution you'd like

It would be nice if a new **component** and **instancing** system can be implemented inside elementor.

![SCR-20230621-psee](https://github.com/elementor/elementor/assets/102000469/c50a2715-eb07-4b47-a18a-936b0fa355f1)

This would solve the following:
- Less custom CSS needed for widgets
- Update is only needed to be applied for the main component and all instances will update

The new component and instancing would be great if paired with "Custom Styles"
<img width="142" alt="SCR-20230621-purj" src="https://github.com/elementor/elementor/assets/102000469/9c6bcf07-7d0f-47a5-859b-71e5e00a071f">
1. Able to save current module changes as a new style
2. Browse saved styles for the selected module without keeping tabs on custom class names

Divi implementation of this one is really great.

I think this is a game changer.

I can imagine creating a hidden page where all components and instances are located which would serve as a styling master dashboard without going inside the elementor editor.

**User Benefits:**
- For new users and clients, they only need to edit this unique page in case they want to update button links, cta sections, or image styles. All their changes will propagate across the site.

- Elementor pages with the new container and nested widgets takes too much browser resources and the peformance quickly drops as the page complexity and size increases. With this, users no longer need to enter visual editor just to update button links, they will be able to see their changes in real-time inside a page with few widget and elements.

- Users can edit elementor style and settings with less RAM usage compared to opened page session with all containers and widgets just sitting in the background.

- Users can update their website using their phone or tablet using their master page. I would call it "Global Styler"

And much more...

### Describe alternatives you've considered

For now, it can be solved by:

**1. Creating custom class**
Pro: 
- Fully customizable

Cons:
- Requires technical know-how for untrained users (client) 
- There's a high chance they could break the cascade specially for media queries. One mistake means broken widget or front-end styling.

**2.Custom Code**
Pro:
- Insteaf of placing all CSS in the Site Settings, I find it more convenient to group related CSS using custom code.
- The editor canvas is large enough which makes it easy to right codes.
- It can accept all codes so I can mix related javascript and CSS.
- I can also target the page and customize how the code gets loaded.
- no need for custom HTML widget

Cons:
- Changes are not shown in realtime (had to use inspect element) but way more convenient compared to the visual editor

**3. Create multiple global widgets**
Pro:
- Can be placed on a single page to serve as master control
- Changes can be seen in real-time

Cons:
-Editing is repetitive and time-consuming
-ACF can be implemented but unecessary

**4. Create multiple styling and store on a separate page**
Pro: 
- You can quickly copy and paste styles as need (not all styling*)

Cons:
- There are times where you cannot copy multiple items at the same time (Elementor error)
- A lot of back and forth between tabs
- Unique classes, attribute, transform or animations get overriden and due to this, it is better to tweak the widget manually to preserve customization–which defeats the purpose (create new instance as needed with unique attributes, transforms etc.)

A few styling limitations:
- Cannot copy background gradient styles
- Cannot copy box shadow styles
-Box shadow cannot use global Colors
-No box Shadow hover effects

### Additional context

I hope Elementor can inherit or take inspiration from the features of Figma and XD in order to improve the user experience of the product. These small features might look insignificant but it's a night and day difference not only for designers but also for developers and specially end-users who are often times unfamulliar about using the product.

**Comment by nicholaszein** [comment]
Hello there, @Henzxon!

Thank you for your detailed submission! 🙏 

### 🔗 🚀 There is already a similar request submitted by someone here #965

Feel free to continue the conversation there! 👉

Kind regards
